app/src/main/java/protect/card_locker/CsvDatabaseImporter.java

```diff
@@ -142,6 +142,8 @@ private void importLoyaltyCard(SQLiteDatabase database, DBHelper helper, CSVReco
             headerTextColor = extractInt(DBHelper.LoyaltyCardDbIds.HEADER_TEXT_COLOR, record, true);
         }
 
-        helper.insertLoyaltyCard(database, id, store, note, cardId, barcodeType, headerColor, headerTextColor);
+        int starStatus = extractInt(DBHelper.LoyaltyCardDbIds.STAR_STATUS, record, false);
```

Comment:
```
2020-07-30 22:00:27.881 29629-29629/protect.card_locker E/LoyaltyCardLocker: Starting file export with: content://com.android.externalstorage.documents/document/primary%3ALoyaltyCardKeychain.csv%20(1)
2020-07-30 22:00:27.960 29629-29718/protect.card_locker E/LoyaltyCardLocker: Failed to input data
    protect.card_locker.FormatException: Field not used but expected: starstatus
        at protect.card_locker.CsvDatabaseImporter.extractInt(CsvDatabaseImporter.java:91)
        at protect.card_locker.CsvDatabaseImporter.importLoyaltyCard(CsvDatabaseImporter.java:145)
        at protect.card_locker.CsvDatabaseImporter.importData(CsvDatabaseImporter.java:32)
        at protect.card_locker.MultiFormatImporter.importData(MultiFormatImporter.java:38)
        at protect.card_locker.ImportExportTask.performImport(ImportExportTask.java:64)
        at protect.card_locker.ImportExportTask.doInBackground(ImportExportTask.java:121)
        at protect.card_locker.ImportExportTask.doInBackground(ImportExportTask.java:16)
        at android.os.AsyncTask$2.call(AsyncTask.java:333)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
```

Should probably fix this, because it sucks if you make a backup but you can't import it in a newer version. Just set starStatus to 0 if the value isn't found in the backup.

Reply:
I would like to suggest solving it this way:

```
diff --git a/app/src/main/java/protect/card_locker/CsvDatabaseImporter.java b/app/src/main/java/protect/card_locker/CsvDatabaseImporter.java
index 5d3f5de..96e8171 100644
--- a/app/src/main/java/protect/card_locker/CsvDatabaseImporter.java
+++ b/app/src/main/java/protect/card_locker/CsvDatabaseImporter.java
@@ -142,8 +142,14 @@ public class CsvDatabaseImporter implements DatabaseImporter
             headerTextColor = extractInt(DBHelper.LoyaltyCardDbIds.HEADER_TEXT_COLOR, record, true);
         }
 
-        int starStatus = extractInt(DBHelper.LoyaltyCardDbIds.STAR_STATUS, record, false);
-        if (starStatus>1 || starStatus<0) throw new FormatException("Wrong Star (Favorite) information listed, but is required");
+        int starStatus = 0;
+        try {
+            starStatus = extractInt(DBHelper.LoyaltyCardDbIds.STAR_STATUS, record, true);
+        } catch (FormatException _) {
+            // This field did not exist in versions 0.28 and before
+            // We catch this exception so we can still import old backups
+        }
+        if (starStatus != 1) starStatus = 0;
         helper.insertLoyaltyCard(database, id, store, note, cardId, barcodeType, headerColor, headerTextColor, starStatus);
     }
 }
```

That way, old backups still work and any invalid value gets parsed as 0.

Reply:
Thanks for the code suggestion! I think this is a very robust and user friendly import now.
In order to avoid NullPointerException a minor change is needed: 
extractInt(DBHelper.LoyaltyCardDbIds.STAR_STATUS, record, **false**);